docker/local-development.md

```diff
@@ -0,0 +1,279 @@
+# Docker for Local Development
+
+Docker for local development was piloted on `dedupe-service`, a repository to
+which some DataMade employees do not have access. Luckily, we can abstract general
+patterns for Dockerizing an application to guide our next effort.
```

Comment:
It'd probably be good to link out to https://github.com/datamade/ihs-website-v2 instead of Dedupe.io here since all DM employees can see it.